_api-reference/index-apis/refresh.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: default
+title: Refresh indexes
+parent: Index APIs
+nav_order: 61
+---
+
+# Example API 
+Introduced 1.0
+{: .label .label-purple }
+
+The Refresh API refreshes one or more indexes in an OpenSearch cluster. In the case of data streams, the Refresh API will refresh a streams backing indexes. 
+
+OpenSearch, by default, will refresh indexes every second on indexes that have received one search request or more in the last 30 seconds. To change the default interval, use the `index.refresh_interval` setting in `opensearch.yml`.
```

Comment:
```suggestion
OpenSearch, by default, refreshes indexes every second for indexes that have received one or more search request in the last 30 seconds. To change the default interval, use the `index.refresh_interval` setting in `opensearch.yml`.
```

Reply:
I don't know a good way to express what's in that diagram as documentation, besides nested if expressions.

Reply:
@msfroh: I think the diagram doesn't help me necessarily, which doesn't mean we shouldn't include it. Is the above suggestion a good understanding of the refresh behavior?

Reply:
How about:

```
- When set, refreshes occur based on the `index.refresh_interval` (in seconds).
- When not set, refreshes will occur every second until the shard receives no search requests for at least `index.search.idle.after` seconds (default 30). Once a shard goes idle, it will not refresh until the next search request or call to the `_refresh` API. Note that the first search request on an idle shard will wait for the refresh operation to complete.
```

I'm also wondering if the Refresh API is the best place to document these details. 

Maybe it would be better to improve the explanation over at https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/index-settings/, and this page could just say:

> OpenSearch, by default, will refresh indexes regularly according to the `index.refresh_interval` and `index.search.idle.after` settings. See the [index settings documentation](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/index-settings/) for more details.

Reply:
I don't think this suggestion was implemented.